### ISSUE TITLE
Question: windows support 

### ISSUE THREAD

**Issue by ivansanchez-oss** [michaelvanstraten/pseudoterminal#3]
In the documentation it says that full support for ``windows`` is not implemented yet, in my case the tests and the example freeze reading from the terminal, is this what the warning was referring to?

nice project!

**Comment by michaelvanstraten** [comment]
Hey, thank you for the comment on this. As mentioned in the readme windows support is planned and it's kind of the only reason the create exists. 

For the time being I could add a compiler warning when using the windows sys backend. 

The implementation is currently stalled due to a wrongly implemented feature in the std library (I am somewhat to blame for this). Maybe you can head over to https://github.com/rust-lang/rust/issues/114854#issuecomment-1872877397 and also comment. I have not received any response from the std lib team.

**Comment by michaelvanstraten** [comment]
https://github.com/rust-lang/rust/pull/121951

**Comment by mnixry** [comment]
> The implementation is currently stalled due to a wrongly implemented feature in the std library (I am somewhat to blame for this).

Hi @michaelvanstraten, Just checking in on this issue regarding Windows support.

I noticed that the Rust PR https://github.com/rust-lang/rust/pull/123604 has been merged into Rust upstream.

Does this merge potentially unblock or make it feasible to complete the Windows pseudo terminal implementation for this crate now?

Thanks for your work on this library!

**Comment by michaelvanstraten** [comment]
> Does this merge potentially unblock or make it feasible to complete the Windows pseudo terminal implementation for this crate now?

I got a WIP implementation ready, but there are still some bugs that I can't debug well in a VM running on arm.

The issue currently is that I am getting unexpected characters from the IO stream, and things like running a shell also don't work well yet. If you want, I can share the WIP and you could also take a look, otherwise I would try to debug it on a real machine.

**Comment by michaelvanstraten** [comment]
This crate should now work on Windows as well. Async support is still lagging behind, but that is an issue for another day. I've published a new version of the crate [here](https://crates.io/crates/pseudoterminal/0.2.1).
